TSError: ⨯ Unable to compile TypeScript ! Sitemap

npm run sitemap http://localhost:3000/

How can i solve this issue ? 
What should i do with ts-node? 


TSError: ⨯ Unable to compile TypeScript:
src/core/utils.ts:132:7 - error TS2322: Type 'ParsedQuery<string>' is not assignable to type 'Record<string, string>'.
  Index signatures are incompatible.
    Type 'string | string[]' is not assignable to type 'string'.
      Type 'string[]' is not assignable to type 'string'.

132   let query: Record<string, string> = parseQs(window.location.search.substr(1));

![Screenshot from 2020-09-07 12-26-27](https://user-images.githubusercontent.com/22500875/92355261-b3174a80-f105-11ea-9fb1-c75efdaf5075.png)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -35,6 +35,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Refactor variant picker components and open sidebar after adding product to cart - #809 by @krzysztofwolski
 - Fix checkout address view - #909 by @konstantinoschristomanos
 - Support for static url - #721 by @marianoeramirez and @dominik-zeglen
+- Fix Generate site map - #915 by @rboixaderg
 
 ## 2.10.4
 

--- a/src/@next/pages/CheckoutPage/CheckoutPage.tsx
+++ b/src/@next/pages/CheckoutPage/CheckoutPage.tsx
@@ -414,7 +414,9 @@ const CheckoutPage: React.FC<IProps> = ({}: IProps) => {
      * https://docs.adyen.com/checkout/drop-in-web?tab=http_get_1#step-6-present-payment-result
      */
     if (
-      adyenNotNegativeConfirmationStatusCodes.includes(querystring.resultCode)
+      adyenNotNegativeConfirmationStatusCodes.includes(
+        querystring.resultCode as string
+      )
     ) {
       const { data, dataError } = await completeCheckout();
       const errors = dataError?.error;
@@ -439,7 +441,7 @@ const CheckoutPage: React.FC<IProps> = ({}: IProps) => {
       setPaymentGatewayErrors([
         {
           message: translateAdyenConfirmationError(
-            querystring.resultCode,
+            querystring.resultCode as string,
             intl
           ),
         },

--- a/src/core/utils.ts
+++ b/src/core/utils.ts
@@ -1,7 +1,11 @@
 import { History, LocationState } from "history";
 import { Base64 } from "js-base64";
 import { each } from "lodash";
-import { parse as parseQs, stringify as stringifyQs } from "query-string";
+import {
+  parse as parseQs,
+  stringify as stringifyQs,
+  ParsedQuery,
+} from "query-string";
 import { FetchResult } from "react-apollo";
 
 import { OrderDirection, ProductOrderField } from "../../gqlTypes/globalTypes";
@@ -128,8 +132,8 @@ export const maybe = <T>(exp: () => T, d?: T) => {
 
 export const parseQueryString = (
   location: LocationState
-): { [key: string]: string } => {
-  let query: Record<string, string> = parseQs(window.location.search.substr(1));
+): ParsedQuery<string> => {
+  let query: ParsedQuery<string> = parseQs(window.location.search.substr(1));
 
   each(query, (value, key) => {
     if (Array.isArray(value)) {

--- a/webpack.d.ts
+++ b/webpack.d.ts
@@ -8,7 +8,6 @@ declare module "*.svg" {
 declare module "draftjs-to-html";
 declare module "react-media";
 declare module "js-base64";
-declare module "query-string";
 
 // This was copied from @saleor/sdk/global.d.ts to make TS compiler happy
 declare interface Window {

